### DISCUSSION TITLE
Whats the behaviour of WintEdt for memorize/learn new commands you define in a document?.

### DISCUSSION THREAD

**Discussion by rafa4mat** [WinEdt-Team/WinEdt-Support#74] [1.1 General Help and Discussion]
Whats the behaviour of WintEdt for memorize/learn newcommand you define in a document?.

For example in the code below, WinEdt after typoe \bol... offers me autocomplete with \boldinred{} BUT not offer me \boldinreds{}{}

```
\documentclass[11pt]{article}
\usepackage{amsmath,mathtools}
\usepackage{pgf}

\newcommand{\boldinred}[1]{{\color{red}\textbf{#1}}}

\newcommand{\boldinreds}[2]{{\color{red}#1 of {\color{red}\textbf{#2}}}}

\begin{document}
Jei
\boldinred{5}\\

Jei jei
\boldinreds{3}{5}

\end{document} 
```

**Comment by Alex-Simonic** [comment]
If the Tree control is visible then rebuilding the tree (F5) updates your commands. Otherwise pressing the ESC at the end of "bol" to manually trigger auto-completion will also update your definitions. It is just a matter of executing the macro `Exe('%b\Macros\Events\DynDict.edt');`. It is available through the Advanced -> Events Section of the Options Interface (should you ever need to customize it). By default, it does not get updated automatically as this could cause delays in large documents. But you can change this in the auto-completion configuration file:

```
[AUTO_COMPLETE]
...
  AUTO_POPUP_EVENT_MACRO=""
  MANUAL_POPUP_EVENT_MACRO="Exe('%b\Macros\Events\DynDict.edt');"
```

Specify `AUTO_POPUP_EVENT_MACRO="Exe('%b\Macros\Events\DynDict.edt');"` and you will always get the latest definitions...

**Reply by rafa4mat** [reply]
Thanks @Alex-Simonic , Works fine.
